src/targets/github.ts

```diff
@@ -277,19 +253,67 @@ export class GitHubTarget extends BaseTarget {
   }
 
   private async handleGitHubUpload(
-    params: RestEndpointMethodTypes['repos']['uploadReleaseAsset']['parameters'],
-    retries = 3
+    release: GitHubRelease,
+    path: string,
+    contentType?: string,
+    retries = 3,
   ): Promise<{ url: string; size: number }> {
+    const contentTypeProcessed = contentType || DEFAULT_CONTENT_TYPE;
+    const stats = statSync(path);
+    const name = basename(path);
+    // this must be recreated each attempt to prevent fd reuse
+    const file = createReadStream(path);
+
+    const params = {
+      ...this.githubConfig,
+      headers: {
+        'Content-Length': stats.size,
+        'Content-Type': contentTypeProcessed,
+      },
+      release_id: release.id,
+      name,
+      // XXX: Octokit types this out as string, but in fact it also
+      // accepts a `Buffer` here. In fact passing a string is not what we
+      // want as we upload binary data.
+      data: file as any,
+      request: {
+        // we are handling retries -- octokit-retries will resuse our fd and
+        // hang forever
```

Comment:
Is there an upstream issue we can make a reference to?

Reply:
the closest I can find is https://github.com/node-fetch/node-fetch/issues/368